MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -3640,6 +3640,20 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 	*pMaxDeviation = cpuEnd - cpuStart;
 }
 
+MVKBuffer* MVKDevice::getBufferAtAddress(uint64_t address)
```

Comment:
This doesn't cut it.

What if the application tries to use a VkBuffer at an offset?
It's perfectly legal to add an offset to the VA as long as its still within a buffer.

Reply:
@K0bin I'm lost on what we would need to allow for offsets when getting the address.

Reply:
Here's what my toy renderer does for example: 
Allocate a single huge VkBuffer for all vertex data in the scene and get the BDA of that.
Then when I build the BLAS, I calculate the VA for where my geometry actually sits which is the BDA + offset.
That then gets passed to the BLAS build. So if you just look for the base BDA of a buffer, you wouldn't find anything in that case.

Reply:
Thanks @K0bin, I almost forgot that I'm storing buffer addresses in a map, so I can't access offset addresses smoothly. My only question is how you get the ending of the address for the buffer?

Reply:
Thanks, I didn't think it was that simple.

Reply:
> EDIT: I read this as "why would we need to allow for offsets".
> 
> Here's what my toy renderer does for example: 
> Allocate a single huge VkBuffer for all vertex data in the scene and get the BDA of that.
> Then when I build the BLAS, I calculate the VA for where my geometry actually sits which is the BDA + offset.
> That then gets passed to the BLAS build. So if you just look for the base BDA of a buffer, you wouldn't find anything in that case.
> 
> For what to do: store buffer ranges and iterate over them to find one that contains the requested address. To speed it up, you could combine this with a map where you mask off some of the lower bits of the VA as a key.

This buffer is just a single giant buffer, which is then suballocated to then accessed via the offset. I'm not sure how we would go about adding these suballocated buffers as regular buffers.